event/oracle/upgrade_oracle_event.go

```diff
@@ -0,0 +1,173 @@
+	queryClient := e.reactor.QueryClient()
+	voterAddress := e.reactor.OracleAcc().GetAddress()
+	oraclePrivKeyBz := e.reactor.OraclePrivKey().Serialize()
+
+	oracleRegistration, err := queryClient.GetOracleRegistration(votingTargetAddress, uniqueID)
+	if err != nil {
+		log.Infof("failed to get oracleRegistration. uniqueID(%s), address(%s). %v", uniqueID, votingTargetAddress, err)
+		return makeMsgVoteOracleRegistrationVoteTypeNo(uniqueID, voterAddress, votingTargetAddress, oraclePrivKeyBz)
+	}
+
+	voteOption, err := e.verifyAndGetVoteOption(oracleRegistration)
+	if err != nil {
+		log.Infof("vote No due to error while verify: %v", err)
+	}
+
+	return makeMsgVoteOracleRegistration(
+		uniqueID,
+		voterAddress,
+		votingTargetAddress,
+		voteOption,
+		oraclePrivKeyBz,
+		oracleRegistration.NodePubKey,
+		oracleRegistration.Nonce,
+	)
+
+}
+
+func (e UpgradeOracleEvent) verifyAndGetVoteOption(oracleRegistration *oracletypes.OracleRegistration) (oracletypes.VoteOption, error) {
+	queryClient := e.reactor.QueryClient()
+	upgradeInfo, err := queryClient.GetOracleUpgradeInfo()
+	if err != nil {
+		return oracletypes.VOTE_OPTION_NO, fmt.Errorf("failed to get oracle upgrade info. %v", err)
+	}
+	if upgradeInfo.UniqueId != oracleRegistration.UniqueId {
+		return oracletypes.VOTE_OPTION_NO, fmt.Errorf("oracle's uniqueID does not match the uniqueID being upgraded. expected uniqueID(%s), oracle's uniqueID(%s), ",
+			upgradeInfo.UniqueId,
+			oracleRegistration.UniqueId,
+		)
+	}
+
+	if err := verifyTrustedBlockInfo(queryClient, oracleRegistration.TrustedBlockHeight, oracleRegistration.TrustedBlockHash); err != nil {
+		return oracletypes.VOTE_OPTION_NO, err
+	}
+
+	if err := e.verifyRemoteReport(oracleRegistration); err != nil {
+		return oracletypes.VOTE_OPTION_NO, fmt.Errorf("failed to verification report. uniqueID(%s), address(%s), err(%v)", oracleRegistration.UniqueId, oracleRegistration.Address, err)
+	}
+
+	return oracletypes.VOTE_OPTION_YES, nil
+
+}
+
+func (e UpgradeOracleEvent) verifyRemoteReport(oracleRegistration *oracletypes.OracleRegistration) error {
+	reportBz := oracleRegistration.NodePubKeyRemoteReport
+	expectedNodePubKeyHash := crypto.KDFSHA256(oracleRegistration.NodePubKey)
+	expectedUniqueID := oracleRegistration.UniqueId
+
+	enclaveInfo := e.reactor.EnclaveInfo()
+	report, err := enclave.VerifyRemoteReport(reportBz)
+	if err != nil {
+		return err
+	}
+
+	if report.SecurityVersion < sgx.PromisedMinSecurityVersion {
+		return fmt.Errorf("invalid security version in the report")
+	}
+	if !bytes.Equal(report.ProductID, enclaveInfo.ProductID) {
+		return fmt.Errorf("invalid product ID in the report")
+	}
+	if !bytes.Equal(report.SignerID, enclaveInfo.SignerID) {
+		return fmt.Errorf("invalid signer ID in the report")
+	}
+	uniqueID, err := hex.DecodeString(expectedUniqueID)
+	if err != nil {
+		return fmt.Errorf("invalid uniqueID format. uniqueID(%s). %w", uniqueID, err)
+	}
+	if !bytes.Equal(report.UniqueID, uniqueID) {
```

Comment:
Should we compare uniqueIDs from report and update info as well?
Maybe three uniqueIDs should be equal..?

Reply:
As I understood your comments, we already checked upgrade's uniqueID and oracleRegistration's uniqueID [here](https://github.com/medibloc/panacea-doracle/pull/91/files#diff-0c6d9bfa1427c39a9489d371dd13d4d5f847f7a562817ee976de39cefadf3bc1R104). :)

Reply:
I missed. Thank you.